interfaces/builtin/dbus_app.go

```diff
@@ -0,0 +1,280 @@
+		snippets.WriteString(``)
+
+		for bus, names := range dbusAppBusNames {
+			// common permanent slot policy
+			abstraction, err := getAppArmorAbstraction(bus)
+			if err != nil {
+				return nil, err
+			}
+			old := []byte("###DBUS_BIND_ABSTRACTION###")
+			new := []byte(abstraction)
+			snippet := bytes.Replace(dbusAppPermanentSlotAppArmorShared, old, new, -1)
+			snippets.Write(snippet)
+
+			for _, name := range names {
+				// well-known DBus name-specific permanent slot
+				// policy
+				snippet := getAppArmorIndividualSnippet(dbusAppPermanentSlotAppArmorIndividual, bus, name)
+				snippets.Write(snippet)
+
+				if release.OnClassic {
+					// classic-only policy
+					snippet := getAppArmorIndividualSnippet(dbusAppPermanentSlotAppArmorIndividualClassic, bus, name)
+					snippets.Write(snippet)
+				}
+			}
+		}
+		//fmt.Printf("DEBUG - PERMANENT SLOT:\n %s\n", snippets.Bytes())
+		return snippets.Bytes(), nil
+	case interfaces.SecuritySecComp:
+		return dbusAppPermanentSlotSecComp, nil
+	case interfaces.SecurityDBus, interfaces.SecurityUDev, interfaces.SecurityMount:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *DbusAppInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus, interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityUDev, interfaces.SecurityMount:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+// Obtain yaml-specified DBus well-known names by bus
+func (iface *DbusAppInterface) getBusNames(attribs map[string]interface{}) (map[string][]string, error) {
+	busNames := make(map[string][]string)
+	for attr := range attribs {
+		bus := attr
+		if bus != "session" && bus != "system" {
+			return nil, fmt.Errorf("bus must be one of 'session' or 'system'")
+		}
+		busNamesList, ok := attribs[bus].([]interface{})
+		if !ok {
+			return nil, fmt.Errorf("bus attribute is not a list")
+		}
+
+		for _, item := range busNamesList {
+			busName, ok := item.(string)
+			if !ok {
+				return nil, fmt.Errorf("session element is not a string")
+			}
+
+			// https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names
+			if !ok || len(busName) == 0 {
+				return nil, fmt.Errorf("bus name must be set")
+			} else if len(busName) > 255 {
+				return nil, fmt.Errorf("bus name is too long (must be <= 255)")
+			}
+
+			validBusName := regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9_-]*(\\.[a-zA-Z0-9_-]+)+$")
+			if !validBusName.MatchString(busName) {
+				return nil, fmt.Errorf("invalid bus name: %q", busName)
+			}
+
+			busNames[bus] = append(busNames[bus], busName)
+		}
+	}
+
+	return busNames, nil
+}
+
+// verify that name for bus is in list
+func (iface *DbusAppInterface) verifyNameInAttributes(bus string, name string, attribs map[string]interface{}) bool {
```

Comment:
No, you are right. This was used in the dbus-bind exploratory PR for making sure the slot side had an entry for the specified plug side. Removed. Thanks!